java/core/src/main/java/com/whylogs/api/logger/resultSets/ResultSet.java

```diff
@@ -0,0 +1,41 @@
+package com.whylogs.api.logger.resultSets;
+
+import com.whylogs.core.DatasetProfile;
+import com.whylogs.core.errors.Error;
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.views.DatasetProfileView;
+import java.util.Optional;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * A holder object for profiling results.
+ *
+ * <p>A whylogs.log call can result in more than one profile. This wrapper class simplifies the
+ * navigation among these profiles.
+ *
+ * <p>Note that currently we only hold one profile but we're planning to add other kinds of profiles
+ * such as segmented profiles here.
```

Comment:
Don't we have segmented profiles already on the Python side?

Reply:
Good question. I copied over this header because it is correct with java not having segments yet. 

As for the python side, yes I think segments are done. I'll check with jamie and send up an update on the python side